src/crimson/os/seastore/seastore_types.h

```diff
@@ -677,6 +676,33 @@ struct blk_paddr_t : public paddr_t {
     return paddr_t::make_blk_paddr(get_device_id(), get_block_off() + o);
   }
 
+  paddr_t add_relative(paddr_t o) const {
+    seastore_off_t off;
+    if (o.get_addr_type() == addr_types_t::SEGMENT) {
+      // segment addr is allocated when alloc_new_extent is called.
+      // But, if random block device is used,
+      // this eventually can be converted to blk addr after submit_record().
```

Comment:
This is misleading, all record-relative or block-relative addrs are represented internally as segmented.  I don't think it's possible to have a non-segmented relative addr.  I think you can ceph_assert(o.get_addr_type() == addr_types_t::SEGMENT).